pkg/coscheduling/coscheduling.go

```diff
@@ -0,0 +1,260 @@
+package coscheduling
+
+import (
+	"context"
+	"math"
+	"strconv"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/api/v1/pod"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+)
+
+// Coscheduling is a plugin that implements the mechanism of gang scheduling.
+type Coscheduling struct {
+	FrameworkHandle framework.FrameworkHandle
+	PodLister       corelisters.PodLister
+	PodGroupInfos   map[string]PodGroupInfo // Key is the name of PodGroup.
+	AutoId          uint64
+}
+
+type PodGroupInfo struct {
+	// InitialTimestamp stores the timestamp of the initialization time of PodGroup.
+	InitialTimestamp time.Time
+	Id               uint64
+	MinAvailable     int
+	Name             string
+}
+
+var _ framework.PreFilterPlugin = &Coscheduling{}
+var _ framework.QueueSortPlugin = &Coscheduling{}
+var _ framework.PermitPlugin = &Coscheduling{}
+var _ framework.UnreservePlugin = &Coscheduling{}
+
+// Name is the name of the plugin used in Registry and configurations.
+const Name = "coscheduling"
+const PodGroupName = "pod-group.scheduling.sigs.k8s.io/name"
+const PodGroupMinAvailable = "pod-group.scheduling.sigs.k8s.io/min-available"
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (cs *Coscheduling) Name() string {
+	return Name
+}
+
+// New initializes a new plugin and returns it.
+func New(_ *runtime.Unknown, handle framework.FrameworkHandle) (framework.Plugin, error) {
+	rand.Seed(time.Now().Unix())
```

Comment:
`rand.Seed()` is a global function and has been called in `main` method, so I think it is redundant here

Reply:
yeah, I find it~ thanks.  I will remove it.